openvdb_ax/openvdb_ax/ast/Parse.cc

```diff
@@ -56,6 +59,8 @@ openvdb::ax::ast::parse(const char* code, openvdb::ax::Logger& logger)
 
     ax_delete_buffer(buffer);
 
+    if (logger.errors() > err) ptr.reset();
```

Comment:
doesnt axparse return 1 if it errors during parsing (https://www.ibm.com/docs/en/zos/2.4.0?topic=works-yyparse-yylex) or is that not 1 in your example case? this could be used to determine erroring instead

Reply:
It does in every case except for right precedence operators - I'm not entirely sure why yet, I suspect its due to error recovering being able to fall back and create the partial AST.